src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -302,28 +301,30 @@
-                                                                 :invalidInputs {}
-                                                                 :validInputs {}
-                                                                 :invalidOutputs {}
-                                                                 :validOutputs {})
-                                                :inputs-outputs (get-parsed-response))
-                                         (swap! state assoc :error (:message (get-parsed-response))))))})
-                        (do
-                          (swap! state assoc :blocker nil :wdl-parse-error (:message response))
-                          (comps/push-error (style/create-server-error-message (:message response)))))))})))})
+         {:endpoint (endpoints/create-template method-ref)
+          :payload method-ref
+          :headers utils/content-type=json
+          :on-done (fn [{:keys [success? get-parsed-response]}]
+                     (let [response (get-parsed-response)]
+                       (if success?
+                         (endpoints/call-ajax-orch
+                           {:endpoint (endpoints/get-validated-workspace-method-config (:workspace-id props) (:config-id props))
+                            :on-done (fn [{:keys [success? get-parsed-response status-text]}]
+                                       (let [valid-response (get-parsed-response)]
+                                         (if success?
+                                           (endpoints/call-ajax-orch
+                                             {:endpoint endpoints/get-inputs-outputs
+                                              :payload (:methodRepoMethod response)
+                                              :headers utils/content-type=json
+                                              :on-done (fn [{:keys [success? get-parsed-response]}]
+                                                         (swap! state dissoc :blocker :wdl-parse-error)
+                                                         (let [template {:methodConfiguration (merge valid-response config-namespace+name)}]
+                                                           (if success?
+                                                               (swap! state assoc
+                                                                      :loaded-config valid-response
+                                                                      :inputs-outputs (get-parsed-response)))
+                                                             (swap! state assoc :error (:message (get-parsed-response)))))})
+                                           (swap! state assoc :error status-text))))})
+                         (do
+                           (swap! state assoc :blocker nil :wdl-parse-error (:message response))
+                           (comps/push-error (style/create-server-error-message (:message response)))))))})))})
```

Comment:
In this code block, I added the get-validated-workspace-method-config call to fix the following bug:
If you edit a method config and then hit "Cancel" error messages underneath empty inputs disappear. (Incidentally, refreshing the page restores the error messages, but they should be there underneath empty required inputs if you cancel the edit mode)